tests/test_eval_detection.py

```diff
@@ -474,3 +511,47 @@ def test_AP_on_filtered_instances() -> None:
     metrics = evaluator.evaluate()
 
     assert metrics.AP.loc["Vehicle"] == 1.0
+
+
+def test_rank() -> None:
+    """Test ranking of detections and scores during detection evaluation."""
+    dts: np.ndarray = np.array(
+        [
+            ObjectLabelRecord(
+                quaternion=np.array([1, 0, 0, 0]),
+                translation=np.array([0, 0, 0]),
+                length=5.0,
+                width=5.0,
+                height=5.0,
+                occlusion=0,
+                score=0.7,
+                track_id="0",
+            ),
+            ObjectLabelRecord(
+                quaternion=np.array([1, 0, 0, 0]),
+                translation=np.array([10, 10, 10]),
+                length=5.0,
+                width=5.0,
+                height=5.0,
+                occlusion=0,
+                score=0.9,
+                track_id="1",
+            ),
+            ObjectLabelRecord(
+                quaternion=np.array([1, 0, 0, 0]),
+                translation=np.array([20, 20, 20]),
+                length=5.0,
+                width=5.0,
+                height=5.0,
+                occlusion=0,
+                score=0.8,
+                track_id="2",
+            ),
+        ]
+    )
+
+    ranked_dts, ranked_scores = rank(dts)
+    track_ids = np.array([dt.track_id for dt in ranked_dts.tolist()])
+    expected_track_ids = np.array(["1", "2", "0"])
+    expected_scores = np.array([0.9, 0.8, 0.7])
+    assert (track_ids == expected_track_ids).all() and (ranked_scores == expected_scores).all()
```

Comment:
[nit]: Just a readability comment: How do you feel like doing?
`all((track_ids == expected_track_ids))` as that is more readable than what we have currently.